src/main/java/org/gridsuite/modification/server/service/NetworkModificationObserver.java

```diff
@@ -26,4 +39,11 @@ private Observation createObservation(String name, ModificationType modification
             .lowCardinalityKeyValue(MODIFICATION_TYPE_TAG_NAME, modificationType.name());
     }
 
+    public void createThreadPoolMetric(ThreadPoolExecutor threadPoolExecutor) {
+        MultiGauge multiGauge = MultiGauge.builder(TASK_POOL_METER_NAME).description("The number of large network modifications (tasks) in the thread pool").register(meterRegistry);
+        multiGauge.register(List.of(
+            MultiGauge.Row.of(Tags.of(TASK_TYPE_TAG_NAME, TASK_TYPE_TAG_VALUE_CURRENT), threadPoolExecutor::getActiveCount),
+            MultiGauge.Row.of(Tags.of(TASK_TYPE_TAG_NAME, TASK_TYPE_TAG_VALUE_PENDING), () -> threadPoolExecutor.getQueue().size())
+        ));
+    }
```

Comment:
I'm not sure to understand why we need a multigauge here as the number of tags is constant and equal to 2. From my understanding multigauge are useful if the number of tags vary.
Here, it seems simpler to have two gauges (one for current, one for pending)?

Reply:
Wiith Jauge.builder we can't have a value supplier different for different tags !

Reply:
```suggestion
    public void createThreadPoolMetric(ThreadPoolExecutor threadPoolExecutor) {
        Gauge.builder(TASK_POOL_METER_NAME + TASK_TYPE_TAG_VALUE_CURRENT, threadPoolExecutor, ThreadPoolExecutor::getActiveCount)
                .description("The number of active diagram generation tasks in the thread pool")
                .tag(TASK_TYPE_TAG_NAME, TASK_TYPE_TAG_VALUE_CURRENT)
                .register(meterRegistry);

        Gauge.builder(TASK_POOL_METER_NAME + TASK_TYPE_TAG_VALUE_PENDING, threadPoolExecutor, executor -> executor.getQueue().size())
                .description("The number of pending diagram generation tasks in the thread pool")
                .tag(TASK_TYPE_TAG_NAME, TASK_TYPE_TAG_VALUE_PENDING)
                .register(meterRegistry);
    }
```
this does not work ?

Reply:
Done